Tasks/UniversalPackagesV0/Common/ArtifactToolUtilities.ts

```diff
@@ -0,0 +1,216 @@
+        toolPath = path.join(dirName, "artifacttool");
+    }
+    return toolPath;
+}
+
+function _createExtractFolder(dest?: string): string {
+    if (!dest) {
+        // create a temp dir
+        dest = path.join(tl.getVariable("Agent.TempDirectory"), "artifactTool");
+    }
+    tl.mkdirP(dest);
+    return dest;
+}
+
+export async function extractZip(file: string): Promise<string> {
+    if (!file) {
+        throw new Error("parameter 'file' is required");
+    }
+    let dest = _createExtractFolder();
+    let zip = new AdmZip(file);
+    zip.extractAllTo(dest, true);
+    return dest;
+}
+
+export async function getArtifactToolFromService(serviceUri: string, accessToken: string, toolName: string){
+
+    let osName = tl.osType();
+    let arch = os.arch();
+    if(osName === "Windows_NT"){
+        osName = "windows";
+    }
+    if (arch === "x64"){
+        arch = "amd64";
+    }
+    const blobstoreAreaName = "clienttools";
+    const blobstoreAreaId = "187ec90d-dd1e-4ec6-8c57-937d979261e5";
+    const ApiVersion = "5.0-preview";
+
+    const credentialHandler = vsts.getBasicHandler("fakeUsername", accessToken);
+    const blobstoreConnection = new vsts.WebApi(serviceUri, credentialHandler);
+
+    let blobstoreApi = blobstoreConnection.getCoreApi();
+
+    // Get packaging versions url
+    const artifactToolGetUrl = await new Promise<string>((resolve, reject) => {
+        let getToolDataPromise = blobstoreApi.vsoClient.getVersioningData(ApiVersion,
+            blobstoreAreaName, blobstoreAreaId, { toolName }, {osName, arch});
+        getToolDataPromise.then((result) => {
+            return resolve(result.requestUrl);
+        });
+        getToolDataPromise.catch((error) => {
+            return reject(error);
+        });
+    });
+
+    // Get highest version
+    const artifactToolUri =  await new Promise<any>((resolve, reject) => {
+        blobstoreApi.restClient.get(artifactToolGetUrl, ApiVersion, null, { responseIsCollection: false },
+            async function (error, status, toolResult) {
+                if (!!error || status !== 200) {
+                    return reject(error.message);
+                }
+                // Getting the tool uri
+                return resolve(toolResult);
+            });
+    });
+
+    let artifactToolPath = toollib.findLocalTool(toolName, artifactToolUri.version);
+    if (!artifactToolPath) {
+        tl.debug(tl.loc("Info_DownloadingArtifactTool", artifactToolUri.uri));
+
+        const zippedToolsDir: string = await toollib.downloadTool(artifactToolUri.uri);
+
+        const unzippedToolsDir = await extractZip(zippedToolsDir);
+
+        artifactToolPath = await toollib.cacheDir(unzippedToolsDir, "ArtifactTool", artifactToolUri.version);
+    }
+    else{
+        tl.debug(tl.loc("Info_ResolvedToolFromCache", artifactToolPath));
+    }
+    return getArtifactToolLocation(artifactToolPath);
+}
+
+// set the console code page to "UTF-8"
+export function setConsoleCodePage() {
```

Comment:
Why is this necessary / is this something every task does?

Reply:
our nuget and dotnet task does it, I don't think it's necessary though. Copy pasted as a boiler plate.

Reply:
I would confirm it's needed / avoid bloat if possible.

Reply:
It was necessary at one point. They may have changed the build agent to do this though